src/openms/include/OpenMS/ANALYSIS/ID/AScore.h

```diff
@@ -78,31 +78,81 @@ class OPENMS_DLLAPI AScore
 
         @note the original sequence is saved in the PeptideHits as MetaValue Search_engine_sequence.
     */
-    PeptideHit compute(PeptideHit & hit, RichPeakSpectrum &real_spectrum, double fragment_mass_tolerance, bool fragment_mass_unit_ppm) const;
-
+    PeptideHit compute(const PeptideHit & hit, RichPeakSpectrum &real_spectrum, double fragment_mass_tolerance, bool fragment_mass_unit_ppm) const;
+
+  protected:
+  
+    /// getSpectrumDifference_ works similar as the method std::set_difference (http://en.cppreference.com/w/cpp/algorithm/set_difference). 
+    /// set_difference was reimplemented, because it was necessary to overwrite the compare operator to be able to compare the m/z values.
+    template <class InputIterator1, class InputIterator2, class OutputIterator>
+    OutputIterator getSpectrumDifference_(InputIterator1 first1, InputIterator1 last1,
+      InputIterator2 first2, InputIterator2 last2, OutputIterator result) const
+    {
+      while (first1 != last1 && first2 != last2)
+      { 
+        if (round(first1->getMZ()) < round(first2->getMZ())) 
```

Comment:
Sorry, it was my mistake. It should be rounded to two decimal places. I will correct this.